source/3ds/gui_hard.c

```diff
@@ -31,8 +31,57 @@
 #define TINT_G ( (tVBOpt.TINT & 0x0000FF00) >> 8 )
 #define TINT_B ( (tVBOpt.TINT & 0x00FF0000) >> 16 )
 
+#define BLACK ( C2D_Color32(0, 0, 0, 255) )
+#define TINT_BRIGHTNESS(BRIGHTNESS) ( C2D_Color32(TINT_R * ( BRIGHTNESS ), TINT_G * ( BRIGHTNESS ), TINT_B * ( BRIGHTNESS ), 255) )
+#define TINT_100 TINT_BRIGHTNESS(1.0)
+#define TINT_90 TINT_BRIGHTNESS(0.9)
+#define TINT_75 TINT_BRIGHTNESS(0.75)
+#define TINT_50 TINT_BRIGHTNESS(0.5)
+#define TINT_COLOR TINT_100
+
+#define PERFORM_FOR_EACH_3DS_BUTTON(MACRO) \
```

Comment:
In order to stop this file's/PR's line count from ballooning too much, we employ some forbidden dark magic here... 😇